exist-core/src/test/java/org/exist/xquery/functions/request/PatchTest.java

```diff
@@ -0,0 +1,164 @@
+
+        bin = (BinaryResource)root.createResource(XQUERY_FILENAME, BinaryResource.RESOURCE_TYPE);
+        ((EXistResource) bin).setMimeType("application/xquery");
+        bin.setContent("xquery version \"3.1\";" +
+                "declare namespace output = \"http://www.w3.org/2010/xslt-xquery-serialization\";" +
+                "declare option output:method \"xml\";" +
+                "declare option output:media-type \"application/xml\";" +
+                "declare option output:indent \"no\";" +
+                "<request><method>{request:get-method()}</method><data>{request:get-data()}</data></request>");
+        root.storeResource(bin);
+        ums.chmod(bin, 0777);
+
+        xml = (XMLResource)root.createResource(XML_FILENAME, XMLResource.RESOURCE_TYPE);
+        xml.setContent("<root/>");
+        root.storeResource(xml);
+        ums.chmod(xml, 0777);
+    }
+
+    @AfterClass
+    public static void afterClass() throws XMLDBException {
+        root.removeResource(bin);
+        root.removeResource(xml);
+    }
+
+    @Test
+    public void patchBinary() throws IOException {
+        final byte[] testData = "12345".getBytes(UTF_8);
+
+        Request patch = Request.Patch(getCollectionRootUri() + "/" + XQUERY_FILENAME)
+                .bodyByteArray(testData, ContentType.APPLICATION_OCTET_STREAM);
+
+        assertResponse(patch, encodeBase64String(testData));
+    }
+
+    @Test
+    public void patchXml() throws IOException {
+        final String testData = "<a><b><c>hello</c></b></a>";
+
+        Request patch = Request.Patch(getCollectionRootUri() + "/" + XQUERY_FILENAME)
+                .bodyByteArray(testData.getBytes(UTF_8), ContentType.TEXT_XML);
+
+        assertResponse(patch, testData);
+    }
+
+    @Test
+    public void patchString() throws IOException {
+        final String testData = "12345";
+
+        Request patch = Request.Patch(getCollectionRootUri() + "/" + XQUERY_FILENAME)
+                .bodyByteArray(testData.getBytes(UTF_8));
+
+        assertResponse(patch, testData);
+    }
+
+    @Test
+    public void patchCollectionNotAllowed() throws IOException {
+        final String testData = "<a><b><c>hello</c></b></a>";
+
+        Request patch = Request.Patch(getCollectionRootUri())
+                .bodyByteArray(testData.getBytes(UTF_8), ContentType.TEXT_XML);
+
+        assertMethodNotAllowed(patch);
+    }
+
+    @Test
+    public void patchXmlResourceNotAllowed() throws IOException {
+        final String testData = "<a><b><c>hello</c></b></a>";
+
+        Request patch = Request.Patch(getCollectionRootUri() + "/" + XML_FILENAME)
+                .bodyByteArray(testData.getBytes(UTF_8), ContentType.TEXT_XML);
+
+        assertMethodNotAllowed(patch);
+    }
+
+    private void assertResponse(Request method, String expectedData) throws IOException {
+        final HttpResponse response = method.execute().returnResponse();
```

Comment:
Do these HttpClients need to be closed after use?

Reply:
Might be the case, IDK. The test class that served as a template did not close them => `GetDataTest`.